akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -477,8 +461,13 @@ private[akka] class RemoteActorRef private[akka] (
 
   def sendSystemMessage(message: SystemMessage): Unit =
     try {
-      remote.send(message, None, this)
-      provider.afterSendSystemMessage(message)
+      message match {
+        case Watch(watchee, watcher) if watcher != provider.remoteWatcher ⇒
+          provider.remoteWatcher ! RemoteWatcher.WatchRemote(watchee, watcher)
```

Comment:
Exactly. this is to improve the case where different local actors are watching the same remote actor.
See discussion here: https://groups.google.com/d/msg/akka-dev/ao6MLKvxJ0o/INMBcgJ7PXMJ